sys/net/gnrc/network_layer/ipv6/nib/_nib-internal.h

```diff
@@ -0,0 +1,476 @@
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @addtogroup  net_gnrc_ipv6_nib
+ * @internal
+ * @{
+ *
+ * @file
+ * @brief       Internal definitions
+ *
+ * @author      Martine Lenders <m.lenders@fu-berlin.de>
+ */
+#ifndef GNRC_IPV6_NIB_INTERNAL_H_
+#define GNRC_IPV6_NIB_INTERNAL_H_
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "kernel_types.h"
+#include "mutex.h"
+#include "net/eui64.h"
+#include "net/ipv6/addr.h"
+#include "net/gnrc/ipv6/nib/nc.h"
+#include "net/gnrc/ipv6/nib/conf.h"
+#include "net/gnrc/pktqueue.h"
+#include "net/ndp.h"
+#include "random.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @{
+ * @brief   Mode flags for entries
+ */
+#define _EMPTY  (0x00)      /**< empty */
+#define _NC     (0x01)      /**< neighbor cache */
+#define _DC     (0x02)      /**< destination cache */
+#define _PL     (0x04)      /**< prefix list */
+#define _DRL    (0x08)      /**< default router list */
+#define _FT     (0x10)      /**< forwarding table */
+#define _DAD    (0x20)      /**< 6LoWPAN duplicate address detection table */
+/** @} */
+
+/**
+ * @brief   Shorter name for convenience ;-)
+ */
+#define _NIB_IF_MASK        (GNRC_IPV6_NIB_NC_INFO_IFACE_MASK)
+
+/**
+ * @brief   Shorter name for convenience ;-)
+ */
+#define _NIB_IF_POS         (GNRC_IPV6_NIB_NC_INFO_IFACE_POS)
+
+/**
+ * @brief   On-link NIB entry
+ */
+typedef struct _nib {
+    struct _nib *next;                  /**< next removable entry */
+#if GNRC_IPV6_NIB_CONF_QUEUE_PKT || defined(DOXYGEN)
+    /**
+     * @brief   queue for packets currently in address resolution
+     *
+     * @note    Only available if @ref GNRC_IPV6_NIB_CONF_QUEUE_PKT != 0
+     */
+    gnrc_pktqueue_t *pktqueue;
+#endif
+    /**
+     * @brief Neighbors IPv6 address
+     */
+    ipv6_addr_t ipv6;
+#if GNRC_IPV6_NIB_CONF_6LN || defined(DOXYGEN)
+    /**
+     * @brief   The neighbors EUI-64 (used for DAD)
+     *
+     * @note    Only available if @ref GNRC_IPV6_NIB_CONF_6LN != 0
+     */
+    eui64_t eui64;
+#endif
+#if GNRC_IPV6_NIB_CONF_ARSM || defined(DOXYGEN)
+    /**
+     * @brief   Link-layer address of _nib_t::next_hop
+     *
+     * @note    Only available if @ref GNRC_IPV6_NIB_CONF_ARSM != 0
+     */
+    uint8_t l2addr[GNRC_IPV6_NIB_L2ADDR_MAX_LEN];
+#endif
+
+    /**
+     * @brief   Information flags (see @ref net_gnrc_ipv6_nib_nc_info "NC info values")
+     */
+    uint16_t info;
+    uint8_t mode;                       /**< NIB mode */
+#if GNRC_IPV6_NIB_CONF_ARSM || defined(DOXYGEN)
+    /**
+     * @brief   Neighbor solicitations sent for probing
+     *
+     * @note    Only available if @ref GNRC_IPV6_NIB_CONF_ARSM != 0
+     */
+    uint8_t ns_sent;
+
+    /**
+     * @brief   length of _nib_t::l2addr
+     *
+     * @note    Only available if @ref GNRC_IPV6_NIB_CONF_ARSM != 0
+     */
+    uint8_t l2addr_len;
```

Comment:
I think the name already reflects that very well (and is used similarly already in other places, e.g. `gnrc_netif_hdr_t`), but I can update the documentation.